### DISCUSSION TITLE
Import execution

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#11274] [General]
If an Import is in a programmed loop, is it only executed once?

MM

**Comment by davefes** [comment]
This didn't get answered:
https://github.com/orgs/micropython/discussions/11199

I am sure I read something posted I think by jimmo about this issue in the last 6 months.

A Lora package I am working with ... the author seems convinced that Micropython uses "lazy loading" unlike regular Python.

Expert answer required!

**Reply by peterhinch** [reply]
It is loaded once only. If you're unsure it could easily be proved by experiment. Create a Python file containing
```py
print("Import")
```
and repeatedly import it.

**Reply by mmliam** [reply]
Peter, I suspected it was loaded only once. It answers an anomaly I've seen in a processor. On boot-up a GPIO goes high until the processor goes into hibernation.
On initial power-up the GPIO stays high noticeably longer than it does on subsequent wake-ups, as imports are initially loaded. The subsequent wake-ups don't reload the imports.

Thanks for the confirmation, 
MM
